src/master/http.cpp

```diff
@@ -2033,11 +2033,26 @@ Future<Response> Master::Http::redirect(const Request& request) const
   // http://stackoverflow.com/questions/12436669/using-protocol-relative-uris-within-location-headers
   // which discusses this.
   string basePath = "//" + hostname.get() + ":" + stringify(info.port());
```

Comment:
These should probably be const strings, which should probably be initialized by constructor rather than assignment.